corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -692,40 +694,43 @@ function GeoPointEntry(question, options) {
     };
 
     self.gMapsCallback = function () {
-        self.geocoder = new google.maps.Geocoder();
-        self.map = new google.maps.Map($('#' + self.entryId)[0], {
-            mapTypeId: google.maps.MapTypeId.ROADMAP,
-            center: new google.maps.LatLng(self.DEFAULT.lat, self.DEFAULT.lon),
-            zoom: self.DEFAULT.zoom,
-        });
-        if (self.rawAnswer().length) {
-            self.map.setCenter(new google.maps.LatLng(self.rawAnswer()[0], self.rawAnswer()[1]));
-            self.map.setZoom(self.DEFAULT.anszoom);
-        }
-        google.maps.event.addListener(self.map, "center_changed", self.updateCenter.bind(self));
-        var marker = new google.maps.Marker({
-            map: self.map,
-        });
-        marker.bindTo('position', self.map, 'center');
+        self.map = L.map(self.entryId).setView([self.DEFAULT.lat, self.DEFAULT.lon], self.DEFAULT.zoom);
+        L.tileLayer(`https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=${window.MAPBOX_ACCESS_TOKEN}`, {
+            id: 'mapbox/streets-v11',
+        }).addTo(self.map);
+        self.map.on('move', self.updateCenter);
+
+        self.centerMarker = L.marker(self.map.getCenter()).addTo(self.map);
+
+        L.mapbox.accessToken = window.MAPBOX_ACCESS_TOKEN;
+        self.geocoder = L.mapbox.geocoder('mapbox.places');
     };
 
     self.afterRender = function () {
-        if (typeof google === "undefined" && !window.gMapsRequested) {
-            // First entry to attempt to load google
-            window.gMapsRequested = true;
-            $.getScript(self.apiKey, self.gMapsCallback);
-        } else if (typeof google === "undefined" && window.gMapsRequested) {
-            // Waiting for gmaps to load, recursively call afterRender
-            setTimeout(self.afterRender, 400);
+        if (typeof L === 'undefined') {
+            // Get the leaflet css, leaflet js, then mapbox js (for the geocoder).
```

Comment:
This is really weird logic to me but I was following the convention of lazy loading the JS lib assets. It requires an ordering of the assets (["Include Leaflet JavaScript file after Leaflet’s CSS"](https://leafletjs.com/examples/quick-start/)). The mapbox JS and CSS actually comes with a [version of leaflet](https://docs.mapbox.com/mapbox.js/api/v3.3.1/standalone/) (pinned to an older version, but still works), but when I used this method to fetch the mapbox js and css alone without the leaflet, it caused icon issues. 

If we added the script and css as intended in the header (probably [here](https://github.com/dimagi/commcare-hq/blob/2b403a5732c781c20631dd20555e9dedec48d45f/corehq/apps/cloudcare/templates/cloudcare/formplayer_home.html#L14)?) it would prevent these issues. Thoughts?

Reply:
I don't know why these scripts are loaded lazily; using script tags sounds better (maybe with `async` if they're slow). Do they definitely need to go into the header? In the page you linked, there's also a `block js` which ends up at the end of the page: https://github.com/dimagi/commcare-hq/blob/27e7afa0975c22e88915d101b0676e41edc440e3/corehq/apps/hqwebapp/templates/hqwebapp/base.html#L547

I'd also be in favor of moving any scripts possible to real package management. @biyeun is currently migrating HQ from bower to yarn, so this probably isn't the best time to do that, but it might be a good followup PR.

Reply:
I ended up putting them [here](https://github.com/dimagi/commcare-hq/pull/28059/commits/3752b0001dd2bffdf36c4aac6d92fa2fc3e459ff#diff-1b662390b8a9b1b43d829121c115fb18R4-R5) to show that they were external dependencies, and also included the css and js together since they're coupled. Does that work?

Reply:
Yep, that looks fine.